vcenter_folder use same parent folder selection

evertmulder opened the pull request:
##### SUMMARY
A different parent folder selection is used for state absent and for state present. The state present allow specify a full path for the folder for a better folder selection. This method is reused for folder removal. 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
Bugfix Pull Request

##### COMPONENT NAME
vcenter_folder

##### ADDITIONAL INFORMATION
Before when calling the module with
```
...
"parent_folder": "Tenants/DI9",
"folder_type": "vm",
"state": "absent",
...
```
would result in 
```
    "msg": "Parent folder Tenants/DI9 does not exist",
```

This is now fixed of the parent folder `Tenants/DI9` does exist

Commit:
vcenter_folder use same parent folder selection

goneri commented:
recheck

evertmulder commented:
I will look into the pep8 issues

goneri commented:
Thanks @evertmulder, the change makes totally sense indeed, and it looks good to me. Could you update the function tests to make use of these absolute paths. This way we will be able to use the CI catch any potential regression in the future?:
https://github.com/ansible/ansible/blob/devel/test/integration/targets/vcenter_folder/tasks/main.yml

evertmulder commented:
@goneri I have added some test. The integration tests seem to fail, but I am not able to trace them back to the test I have added. Could you review the added tests?

goneri commented:
recheck

goneri commented:
recheck

evertmulder commented:
@meerkampdvv @goneri. It looks to me this PR solves the same issue as https://github.com/ansible/ansible/pull/65283. Since https://github.com/ansible/ansible/pull/65283 is already merged I will close this PR. 
Although the module has lots of code duplication right now for the `absent` and `present` state.
